pug/job_submit.pug

```
@@ -0,0 +1,138 @@
+    
+    // Submit the job to the DB
+    function SubmitData(submission)
+    {
+      try
+      {
+        // Include relevant information about the job type form (these MUST be in the submission information)
+        submission.formId = form.formId;
+        submission.formName = form.formName;
+        submission.formObjectId = form._id;
+        
+        // If the submission is successful ...
+        function postSuccess(retval)
+        {
+          // Just in case, output any error messages that might have occurred even with a successful submission
+          if(retval.error)
+          {
+            jobForm.setAlert("warning", retval.error);
+            jobForm.emit('submitError');
+          }
+          
+          jobForm.emit('submitDone');
+            
+          // Go to the page for viewing information about the submitted job, and do not leave the submission page in the browser history
+          window.location.replace('/job/' + retval);
+        }
+        
+        // If the submission fails ...
+        function postFail(res, statusCode, statusMsg)
+        {
+          // Set up and relay error messages
+          if(res.responseText && res.responseText.length > 0)
+          {
+            jobForm.setAlert("danger", res.responseText);
+          }
+          else
+          {
+            jobForm.setAlert("danger", statusMsg + " (" + statusCode + ")");
+          }
+          
+          jobForm.emit('submitError');
+        }; 
+
+        // Perform the submission to the DB, using the success and failure functions defined above
+        $.ajax({contentType: "application/json",
+                method: "post",
+                url: '/json/job',
+                data: JSON.stringify(submission),
+                dataType: "json",
+                success: postSuccess})
+         .fail(postFail);
+      }
+      catch(e)
+      {
+        // TOFIX: not sure what to put here (used to be console and debugger, but we're no longer using them)
```

If we don’t have a logging solution of any kind maybe we should remove the try…catch statement altogether.  Removing the statement will allow errors to throw, which will be visible in the console but may also break something in the UI.  Leaving the catch in hides the errors.